src/debug/dwarf/line_numbers.cr

```diff
@@ -389,20 +395,56 @@ module Debug
         end
       end
 
-      private def register_filename(name)
-        if index = @files.index(name)
-          return index
+      # When decoding statement programs when asking for the current
+      # filename it's usually the case that it's the same as the old
+      # one (because all info from a single file comes first, then
+      # another file comes next, etc.). So we remember the last
+      # mapped index to avoid unnecessary lookups.
+      @last_filename : String?
+      @last_filename_index = 0
```

Comment:
BTW, if the files are read in some order (and not jumping from file to file, do `@last_filename` ever differ from `@files[-1]`, (and similar for the index)?

Reply:
Yes, they differ. They are not read in order. I don't know why, though, I'm not familiar with the format. Maybe understanding the format first, then optimizing it might be better... 😊

Reply:
If I recall correctly, the format follows functions (symbols). An instruction group of the function refers to a file:line, then the following group likely refers to the same file at a further line... unless the compiler inlined code, which will cause jumps to another file, etc.

If I recall correctly, that is.